core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DiskUpdateController.java

```diff
@@ -0,0 +1,56 @@
+package com.sequenceiq.cloudbreak.controller.v4;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.AccountIdNotNeeded;
+import com.sequenceiq.authorization.annotation.InternalOnly;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.DiskUpdateEndpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.DiskUpdateRequest;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.SdxDiskUpdateRequest;
+import com.sequenceiq.cloudbreak.cloud.model.Volume;
+import com.sequenceiq.cloudbreak.service.datalake.DiskUpdateService;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+
+@Controller
+@InternalOnly
+public class DiskUpdateController implements DiskUpdateEndpoint {
+
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
+    @Override
+    @AccountIdNotNeeded
+    public Boolean isDiskTypeChangeSupported(String platform) {
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
+    }
+
+    @Override
+    @AccountIdNotNeeded
+    public void updateDiskTypeAndSize(SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception {
+        DiskUpdateRequest diskUpdateRequest = sdxDiskUpdateRequest.getDiskUpdateRequest();
+        List<Volume> volumesToUpdate = sdxDiskUpdateRequest.getVolumesToUpdate();
+        diskUpdateService.updateDiskTypeAndSize(diskUpdateRequest, volumesToUpdate, sdxDiskUpdateRequest.getStackId());
+    }
+
+    @Override
+    @AccountIdNotNeeded
+    public void stopCMServices(long stackId) throws Exception {
+        diskUpdateService.stopCMServices(stackId);
```

Comment:
are you sure accountid is not needed in underlying layers?

Reply:
yes, account id is not needed for this flow